cfme/intelligence/reports/reports.py

```diff
@@ -78,8 +79,20 @@ class filter(WaitTab):  # noqa
         interval = BootstrapSelect("cb_interval")
         interval_size = BootstrapSelect("cb_interval_size")
         interval_end = BootstrapSelect("cb_end_interval_offset")
-        primary_filter = ExpressionEditor()
-        secondary_filter = ExpressionEditor()
+        primary_record_filter = Button("contains", "Record Filter")
+        secondary_display_filter = Button("contains", "Display Filter")
+
+        @View.nested
+        class primary_filter(ExpressionEditor):     # noqa
+            def child_widget_accessed(self, widget):
```

Comment:
_Question_ could you explain why `widget` is passed here? I haven't seen this used before and am curious as to how it works.

Reply:
Okay so `child_widget_accessed` is a method that allows us to perform pre-requisite actions before accessing child-widgets of a Widget. In my case, I need to click on `primary_record_filter` to access Primary Filter and `secondary_display_filter` to access Secondary Filter. `child_widget_accessed` clicks those button for me.

The `widget` passed as an argument here is the child_widget we're trying to accessing.

Reply:
Many thanks for the explanation - makes sense! :+1: